### PR TITLE
feat: suggested follows by MadFi

### PR DESCRIPTION
**Pull request by imthatcarlos**
## What does this PR do?

Replaces the component for `Recommended Profiles` in the explore page with `Suggested Follows`
- API request is made to `https://api.madfinance.xyz` (requires api key in header)
- `?profileId` query param is included
- profiles in the response are based on matching ML tags (lens interests)
- if the profile has not been tagged - add them to a queue to be tagged - and return trending profiles
- promoted profiles for a specific tag (lens interest) are shown at the top of the list with a Promoted tag
- when promoted profiles are followed, an impression is logged for lens client rev share

Demo: https://lenstube.xyz/watch/0x21c0-0x67-DA-5bb6c621
API docs: https://docs.madfinance.xyz/api/suggested-follows

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Explanation of the changes

- new component `Suggested Follows` replaces `RecommendedProfiles`
- new package `madfi` for hooks to 1) fetch profiles 2) log impression

## Emoji

💸

### CODE REVIEW

**Comment by bigint** [comment]
Hey thanks for the PR, but can we have those in explore page (like you can create section called explore profiles and plugin there), cuz we don't want to disturb core feature on the home page 🙇🏼

**Comment by imthatcarlos** [comment]
@bigint understood! i reverted the change to the home page. for the explore page, im not sure where to place this new section. the left side seems to be a specific feed for posts, leaving only the right side for profile list. what if we left it there?

Screen Shot 2023-05-31 at 6 45 07 PM

**Comment by bigint** [comment]
@imthatcarlos WDYM `creator eco` in the screenshot? will that change?

File: packages/madfi/utils.ts
Comment (bigint) [inline packages/madfi/utils.ts:1]: we should not break forks and other dev envs, being all GET endpoints why we need API key?

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by bigint** [comment]
PS: CI breaks 😕

File: packages/madfi/utils.ts
Comment (imthatcarlos) [reply packages/madfi/utils.ts:1]: yeah so this is where I was hoping for some guidance. we could put this feature behind a feature flag, and maybe it's best that if we _don't_ have an API key, we just see the recommended profiles component.

to answer your question, this GET request can be expensive if the given `profileId` in the query params has **not** been tagged; we spin up a job that runs our ML pipeline to return matching creators on the next uncached request. Finally, the api key is especially important considering the `/impression` POST request - that helps us with the client rev share.

**Comment by imthatcarlos** [comment]
@bigint - I conditionally render the new component if there is an api key in the env. Also, merged in `main` which had a lot of changes since yesterday. Finally, not sure what's going on with the CI, it always says `waiting for status`

**Comment by bigint** [comment]
Closing this, we can revisit this idea later 🙇🏼

**Pull request closed without merging**